src/test/java/io/managed/services/test/rhoas/KafkaRhoasAclTests.java

```diff
@@ -0,0 +1,367 @@
+
+        try {
+            cli.deleteAllAcls(serviceAccount);
+        } catch (Throwable t) {
+            LOGGER.error("delete ACLs error: ", t);
+        }
+
+        try {
+            KafkaMgmtApiUtils.deleteKafkaByNameIfExists(kafkaMgmtApi, KAFKA_INSTANCE_NAME);
+        } catch (Throwable t) {
+            LOGGER.error("delete kafka instance error: ", t);
+        }
+
+        try {
+            SecurityMgmtAPIUtils.cleanServiceAccount(securityMgmtApi, SERVICE_ACCOUNT_NAME);
+        } catch (Throwable t) {
+            LOGGER.error("delete service account error: ", t);
+        }
+
+        try {
+            LOGGER.info("logout user from rhoas");
+            cli.logout();
+        } catch (Throwable t) {
+            LOGGER.error("CLI logout error: ", t);
+        }
+
+        try {
+            LOGGER.info("delete workdir: {}", cli.getWorkdir());
+            FileUtils.deleteDirectory(new File(cli.getWorkdir()));
+        } catch (Throwable t) {
+            LOGGER.error("clean workdir error: ", t);
+        }
+
+        bwait(vertx.close());
+    }
+
+    @SneakyThrows
+    public void downloadCLI() {
+
+        var downloader = CLIDownloader.defaultDownloader();
+
+        // download the cli
+        var binary = downloader.downloadCLIInTempDir();
+
+        this.cli = new CLI(binary);
+
+        LOGGER.info("validate cli");
+        LOGGER.debug(cli.help());
+    }
+
+    @SneakyThrows
+    public void login() {
+
+        // make sure you are logout while developing locally.
+        LOGGER.info("verify that we aren't logged-in");
+        assertThrows(CliGenericException.class, () -> cli.listKafka());
+
+        LOGGER.info("login the CLI");
+        CLIUtils.login(vertx, cli, Environment.PRIMARY_USERNAME, Environment.PRIMARY_PASSWORD).get();
+
+        LOGGER.info("verify that we are logged-in");
+        cli.listKafka();
+    }
+
+    @SneakyThrows
+    public void createServiceAccount() {
+
+        LOGGER.info("create a service account");
+        serviceAccountSecret = CLIUtils.createServiceAccount(cli, SERVICE_ACCOUNT_NAME);
+
+        LOGGER.info("get the service account");
+        var sa = CLIUtils.getServiceAccountByName(cli, SERVICE_ACCOUNT_NAME);
+        LOGGER.debug(sa);
+
+        assertTrue(sa.isPresent());
+        assertEquals(sa.get().getName(), SERVICE_ACCOUNT_NAME);
+        assertEquals(sa.get().getClientId(), serviceAccountSecret.getClientID());
+
+        serviceAccount = sa.get();
+        serviceAccount.setSecret(serviceAccountSecret.getClientSecret());
+    }
+
+    @SneakyThrows
+    public void createKafkaInstance() {
+
+        LOGGER.info("create kafka instance with name {}", KAFKA_INSTANCE_NAME);
+        var k = cli.createKafka(KAFKA_INSTANCE_NAME);
+        LOGGER.debug(k);
+
+        LOGGER.info("wait for kafka instance: {}", k.getId());
+        kafka = CLIUtils.waitUntilKafkaIsReady(cli, k.getId());
+        LOGGER.debug(kafka);
+    }
+
+    @SneakyThrows
+
+    @Test(priority = 1)
```

Comment:
Not necessary here, actually we could get rid of it in this case completly.

Reply:
It's necessary to set priority. If not, all createAcl() tests run first and the all deleteAcl() tests run at the end. This may create conflict trying to delete the same ACL more than once.

Reply:
yes but it would be probably better to go with `dependsOn` annotation for corresponding creation/deletion tests.